GVFS/GVFS.Service/Handlers/GetActiveRepoListHandler.cs

```diff
@@ -69,23 +69,31 @@ public void Run()
         private bool IsValidRepo(string repoRoot)
         {
             string gitBinPath = GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath();
-            string hooksPath = ProcessHelper.WhereDirectory(GVFSPlatform.Instance.Constants.GVFSHooksExecutableName);
-            GVFSEnlistment enlistment = null;
+            string hooksPath = null;
 
-            try
+            string hooksVersion = null;
+            string error = null;
+            if (GVFSPlatform.Instance.TryGetGVFSHooksPathAndVersion(out hooksPath, out hooksVersion, out error))
             {
-                enlistment = GVFSEnlistment.CreateFromDirectory(repoRoot, gitBinPath, hooksPath, authentication: null);
-            }
-            catch (InvalidRepoException e)
-            {
-                EventMetadata metadata = new EventMetadata();
-                metadata.Add(nameof(repoRoot), repoRoot);
-                metadata.Add(nameof(gitBinPath), gitBinPath);
-                metadata.Add(nameof(hooksPath), hooksPath);
-                metadata.Add("Exception", e.ToString());
-                this.tracer.RelatedInfo(metadata, $"{nameof(this.IsValidRepo)}: Found invalid repo");
+                try
+                {
+                    GVFSEnlistment enlistment = null;
+                    if (GVFSPlatform.Instance.TryGetGVFSHooksPathAndVersion(out hooksPath, out hooksVersion, out error))

```

Comment:
Why are we calling `TryGetGVFSHooksPathAndVersion` twice?  What would be different the second time?

Reply:
+1, I don't understand this either.

Reply:
My bad (copy paste error). Removed the redundancy.